### DISCUSSION TITLE
Can I set a independent probability in the RandomAffine?

### DISCUSSION THREAD

**Discussion by Bigsealion** [TorchIO-project/torchio#845] [Q&A] [answered]
I hope the scales, degrees, and translation will be applied in independent probability, but I only find a args `p` in `RandomAffine`, but not like the arg `flip_probability` in `RandomFlip`.

Now, I tried to do this in the following code (I hope this is right):
```
tio.Compose([
    tio.RandomAffine(scales=[0.8, 1.2], degrees=0, translation=0, p=0.5),
    tio.RandomAffine(scales=[1, 1], degrees=30, translation=0, p=0.2),
    tio.RandomAffine(scales=[1, 1], degrees=0, translation=10, p=0.3),
])
```

But I want to know, have any native way to do this?

Thanks!

**Comment by romainVala** [comment, marked as answer]
Hello
no this is not implemented, not sure though why you want this. 
A random affine is a mixture of scaling rotation and translation .... and since rotation for instance randomly choose an angle between 0 and 30, there will be cases were there is almost no rotations

your  proposed solution will work, but then be aware that you are performing 3 interpolation (so more bluring than only one interpolation) (this may be important ... or not depending on the application)

**Reply by Bigsealion** [reply]
Thanks for your reply!
